### PR TITLE
erlfmt: init at 1.0.0

### PR DESCRIPTION
**Pull request by dlesl**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/beam-modules/erlfmt/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  repo = "erlfmt";
+in rebar3Relx {
+  releaseType = "escript";
+  name = "erlfmt";
+  inherit version;
+  src = fetchFromGitHub {
+    inherit owner repo;
+    sha256 = "19apbs9xr4j8qjb3sv9ilknqjw4a7bvp8jvwrjiwvwnxzzm2kjm6";
+    rev = "v${version}";
+  };
```
Comment (SuperSandro2000) [inline pkgs/development/beam-modules/erlfmt/default.nix:15]:
```suggestion

rebar3Relx rec {
  pname = "erlfmt";
  version = "1.0.0";
  releaseType = "escript";

  src = fetchFromGitHub {
    owner = "WhatsApp";
    repo = "erlfmt";
    sha256 = "19apbs9xr4j8qjb3sv9ilknqjw4a7bvp8jvwrjiwvwnxzzm2kjm6";
    rev = "v${version}";
  };
```

File: pkgs/development/beam-modules/erlfmt/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    PATH=${lib.makeBinPath [ common-updater-scripts coreutils git gnused ]}
+    latest=$(list-git-tags https://github.com/${owner}/${repo}.git | sed -n 's/^v//p' | sort -V | tail -1)
+    if [ "$latest" != "${version}" ]; then
+      nixpkgs="$(git rev-parse --show-toplevel)"
+      nix_file="$nixpkgs/pkgs/development/beam-modules/erlfmt/default.nix"
+      update-source-version erlfmt "$latest" --version-key=version --print-changes --file="$nix_file"
+    else
+      echo "erlfmt is already up-to-date"
+    fi
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/beam-modules/erlfmt/default.nix:28]:
```suggestion

```
nix-update should do the same.

Comment (dlesl) [reply pkgs/development/beam-modules/erlfmt/default.nix]: thanks, I didn't know about that!

File: pkgs/development/beam-modules/erlfmt/default.nix [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{ fetchFromGitHub, rebar3Relx, lib }:
+
+rebar3Relx rec {
+  name = "erlfmt";
```
Comment (SuperSandro2000) [inline pkgs/development/beam-modules/erlfmt/default.nix:4]:
```suggestion
  pname = "erlfmt";
```

Comment (dlesl) [reply pkgs/development/beam-modules/erlfmt/default.nix:4]: rebar3Relx doesn't accept `pname`, probably something we should fix in another PR?

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 129250` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>erlfmt</li>
  </ul>
</details>

**Comment by happysalada** [comment]
Looks good to me!
Thank you for this! ❤️

File: pkgs/development/beam-modules/erlfmt/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/development/beam-modules/erlfmt/default.nix:4]: Yeah, didn't remember that. @happysalada could you take a look at this? Everything should accept pname to make it uniform.

**Pull request merged**